app/src/main/java/com/teamagam/gimelgimel/app/view/adapters/MessagesRecyclerViewAdapter.java

```diff
@@ -0,0 +1,133 @@
+package com.teamagam.gimelgimel.app.view.adapters;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.teamagam.gimelgimel.R;
+import com.teamagam.gimelgimel.app.model.ViewsModels.Message;
+
+import java.text.SimpleDateFormat;
+import java.util.Map;
+import java.util.TreeMap;
+
+import butterknife.BindView;
+
+
+/**
+ * {@link RecyclerView.Adapter} that can display a {@link Message} and makes a call to the
+ * specified {@link OnItemClickListener}.
+ */
+public class MessagesRecyclerViewAdapter extends
+        BaseRecyclerArrayAdapter<MessagesRecyclerViewAdapter.MessageViewHolder, Message> {
+
+    private static Map<String, Integer> sTypeMessageMap = new TreeMap<>();
+
+    private static final int TYPE_TEXT = 0;
+    private static final int TYPE_LAT_LONG = 1;
+    private static final int TYPE_IMAGE = 2;
+
+    static {
+        sTypeMessageMap.put(Message.TEXT, TYPE_TEXT);
+        sTypeMessageMap.put(Message.LAT_LONG, TYPE_LAT_LONG);
+        sTypeMessageMap.put(Message.IMAGE, TYPE_IMAGE);
+    }
+
+    private final OnItemClickListener mListener;
+
+    public MessagesRecyclerViewAdapter(MessageListViewModel.DisplayMessagesRandomAccessor accessor, MessagesRecyclerViewAdapter.OnItemClickListener listener) {
+        super(accessor);
+        mListener = listener;
+    }
+
+    @Override
+    protected MessageViewHolder createNewViewHolder(View view) {
+        return new MessageViewHolder(view);
+    }
+
+    @Override
+    protected int getSingleItemLayoutRes() {
+        return R.layout.fragment_messages_list_item;
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        return getMessageType(mAccessor.get(position));
+    }
+
+    private static int getMessageType(Message msg) {
+        return sTypeMessageMap.get(msg.getType());
+    }
+
+    protected void bindItemToView(final MessageViewHolder holder) {
+        drawMessageIcon(holder);
+        drawMessageDate(holder);
+        drawMessageBackground(holder);
+
+        holder.senderTV.setText(holder.item.getSenderId());
```

Comment:
you could have done the same here, abstract the little details of binding the sender TextView data with the viewHolder associated string with a simple method extraction